src/Symfony/Component/Translation/DependencyInjection/TranslatorPass.php

```diff
@@ -19,13 +19,13 @@
 class TranslatorPass implements CompilerPassInterface
 {
     private $translatorServiceId;
-    private $loaderServiceId;
+    private $readerServiceId;
     private $loaderTag;
 
-    public function __construct($translatorServiceId = 'translator.default', $loaderServiceId = 'translation.loader', $loaderTag = 'translation.loader')
+    public function __construct($translatorServiceId = 'translator.default', $readerServiceId = 'translation.reader', $loaderTag = 'translation.loader')
```

Comment:
Sorry, I haven't commented on the right place and my comment was not clear enough.
What I mean is that anyone relying on the deprecated `translation.loader` service must still be able to use it the same as before (for BC). Thus, the compiler pass must still wire it the same as before.

I suggest to:
- Keep `translation.loader` as `$readerServiceId` default value here (this pass is part of the component which can be used in the wild, changing the default value here is a BC break)
- In the constructor, if `'translation.loader' === $readerServiceId` then trigger a deprecation saying that the default value will be `translation.reader` in 4.0 (needs to be documented in UPGRADE-* as well).
- Duplicate the `process()` logic for wiring the new `translation.reader` service (hardcoding the id, reusing all the things that are injected in `$readerServiceId`)

Like this we can just change the default value of `$readerServiceId` to `translation.reader` in 4.0, removing the duplicated code.